Dockerfile

```diff
@@ -0,0 +1,25 @@
+FROM 760097843905.dkr.ecr.eu-west-1.amazonaws.com/node:16-alpine as production
+WORKDIR /app
+
+COPY yarn.lock package.json ./
+RUN yarn install --production --frozen-lockfile && yarn cache clean
+
+COPY tsconfig.json index.ts ./
+COPY src ./src
+
+ENV NODE_ENV=production
+ENV PORT=8000
+
+CMD ["yarn", "start"]
+
+# Use the multi-stage build to install devDependencies separately
+FROM production as testing
```

Comment:
[Multi-stage builds](https://docs.docker.com/develop/develop-images/multistage-build/) are a cool way to get smaller docker images.

Here, we build an absolutely minimal image first, then in the 2nd stage we add in tests and development dependencies: that means we can build the whole thing and run our tests in CI, then only push the "top" layers (without the dev dependencies) to ECR to be run as an application. That - along with using Alpine - means that the image is only 53mb!

The one drawback here is that we have to be careful about making sure the `production` image is functional: we might accidentally add in some _required_ stuff after this line. That's a risk I'm happy to take in general, and certainly happy for a non-prod service like this one.